packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsList.vue

```diff
@@ -23,177 +23,111 @@
 	</div>
 </template>
 
-<script lang="ts">
-import type { PropType } from 'vue';
-import { defineComponent } from 'vue';
-import { mapStores } from 'pinia';
-import { useRouter } from 'vue-router';
+<script setup lang="ts">
+import { computed } from 'vue';
+import { onBeforeRouteLeave, useRouter } from 'vue-router';
 import WorkflowExecutionsSidebar from '@/components/executions/workflow/WorkflowExecutionsSidebar.vue';
-import {
-	MAIN_HEADER_TABS,
-	MODAL_CANCEL,
-	MODAL_CONFIRM,
-	PLACEHOLDER_EMPTY_WORKFLOW_ID,
-	VIEWS,
-} from '@/constants';
+import { MAIN_HEADER_TABS, VIEWS } from '@/constants';
 import type { ExecutionFilterType, IWorkflowDb } from '@/Interface';
-import type { ExecutionSummary, IDataObject } from 'n8n-workflow';
-import { useMessage } from '@/composables/useMessage';
-import { useToast } from '@/composables/useToast';
+import type { ExecutionSummary } from 'n8n-workflow';
 import { getNodeViewTab } from '@/utils/canvasUtils';
 import { useWorkflowHelpers } from '@/composables/useWorkflowHelpers';
 import { useUIStore } from '@/stores/ui.store';
-import { useSettingsStore } from '@/stores/settings.store';
-import { useNodeTypesStore } from '@/stores/nodeTypes.store';
-import { useTagsStore } from '@/stores/tags.store';
-import { executionFilterToQueryFilter } from '@/utils/executionUtils';
-import { useExternalHooks } from '@/composables/useExternalHooks';
-import { useDebounce } from '@/composables/useDebounce';
 import { useNpsSurveyStore } from '@/stores/npsSurvey.store';
+import { watch } from 'vue';
 
-export default defineComponent({
-	name: 'WorkflowExecutionsList',
-	components: {
-		WorkflowExecutionsSidebar,
+const props = withDefaults(
+	defineProps<{
+		loading: boolean;
+		workflow: IWorkflowDb;
+		executions: ExecutionSummary[];
+		execution?: ExecutionSummary;
+		loadingMore: boolean;
+	}>(),
+	{
+		loading: false,
+		executions: () => [] as ExecutionSummary[],
+		loadingMore: false,
 	},
-	async beforeRouteLeave(to, _, next) {
-		if (getNodeViewTab(to) === MAIN_HEADER_TABS.WORKFLOW) {
-			next();
+);
+
+const emit = defineEmits<{
+	'execution:delete': [value?: string];
+	'execution:stop': [value?: string];
+	'execution:retry': [value: { id: string; loadWorkflow: boolean }];
+	'update:auto-refresh': [value: boolean];
+	'update:filters': [value: ExecutionFilterType];
+	'load-more': [];
+	reload: [];
+}>();
+
+const uiStore = useUIStore();
+const npsSurveyStore = useNpsSurveyStore();
+const workflowHelpers = useWorkflowHelpers({ router: useRouter() });
+const router = useRouter();
+
+const temporaryExecution = computed(() => {
+	const isTemporary = !props.executions.find((execution) => execution.id === props.execution?.id);
+	return isTemporary ? props.execution : undefined;
+});
+
+const hidePreview = computed(() => {
+	return props.loading || (!props.execution && props.executions.length);
+});
+
+const onDeleteCurrentExecution = () => {
+	emit('execution:delete', props.execution?.id);
+};
```

Comment:
```suggestion
const onDeleteCurrentExecution = () => {
  if (!props.execution) return;
  
	emit('execution:delete', props.execution.id);
};
```

Add a guard here to fix emit types.